### PR TITLE
Fix NPE when deleting an authorized API

### PR DESCRIPTION
**Pull request by ThaminduR**
### Proposed changes in this pull request

$subject

### CODE REVIEW

File: components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth/listener/IdentityOauthEventHandler.java [hunk 1]
```diff
@@ -230,11 +230,11 @@ public void handleEvent(Event event) throws IdentityEventException {
                 AuthorizedAPI authorizedAPI = OAuthComponentServiceHolder.getInstance()
                         .getAuthorizedAPIManagementService()
                         .getAuthorizedAPI(appId, apiId, tenantDomain);
-                List<String> removedScopes = new ArrayList<>();
-                removedScopes.addAll(authorizedAPI.getScopes().stream()
-                                .map(Scope::getName).filter(scope ->
-                        !removedScopes.contains(scope)).collect(Collectors.toList()));
-
+                if (authorizedAPI.getScopes() == null) {
+                    return;
```
Comment (Thumimku) [inline components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth/listener/IdentityOauthEventHandler.java:234]: can we use isempty

File: components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth/listener/IdentityOauthEventHandler.java [hunk 2]
```diff
@@ -230,11 +230,11 @@ public void handleEvent(Event event) throws IdentityEventException {
...
+                }
+                List<String> removedScopes = authorizedAPI.getScopes().stream()
```
Comment (Thumimku) [inline components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth/listener/IdentityOauthEventHandler.java:236]: Will this handle duplicate scopes, don't we need to check removedScopes.contains(scope) before adding or we can make it as `set`

**Review by SujanSanjula96 — APPROVED** [review]

File: components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth/listener/IdentityOauthEventHandler.java (hunk 1 above)
Comment (ThaminduR) [reply components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth/listener/IdentityOauthEventHandler.java:234]: this object itself can be null

File: components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth/listener/IdentityOauthEventHandler.java (hunk 2 above)
Comment (ThaminduR) [reply components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth/listener/IdentityOauthEventHandler.java:236]: there can't be any duplicated scopes coming in the authorized scopes list

**Pull request merged**